libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -234,12 +234,82 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 
 void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 {
-	if (type(_unaryOperation).category() == Type::Category::RationalNumber)
+	auto const& opType = type(_unaryOperation);
+	Token const op = _unaryOperation.getOperator();
+
+	if (opType.category() == Type::Category::RationalNumber)
+	{
 		defineExpression(_unaryOperation) <<
-			formatNumber(type(_unaryOperation).literalValue(nullptr)) <<
+			formatNumber(opType.literalValue(nullptr)) <<
 			"\n";
+	}
+	else if (opType.category() == Type::Category::Integer)
+	{
+		string intermediateValue = m_context.newYulVariable();
+
+		if (op == Token::Inc || op == Token::Dec)
+		{
+			auto const* intType = dynamic_cast<IntegerType const*>(&opType);
+
+			solAssert(!!m_currentLValue, "LValue not retrieved.");
+			string lvalue = m_currentLValue->retrieveValue();
```

Comment:
Could you please add the following remark to the documentation of `retrieveValue`?

Has to be the name of a variable, i.e. side-effect free.

Reply:
Or instead store it in a variable called `previousValue`.

Reply:
`previousValue` is already used below. I went with `originalValue` now